packages/discord.js/typings/index.d.ts

```diff
@@ -5349,11 +5434,12 @@ export interface MessageReference {
 export type MessageResolvable = Message | Snowflake;
 
 export interface SelectMenuComponentData extends BaseComponentData {
-  type: ComponentType.SelectMenu;
+  type: SelectMenuType;
   customId: string;
   disabled?: boolean;
   maxValues?: number;
   minValues?: number;
+  channelTypes?: ChannelType[];
   options?: SelectMenuComponentOptionData[];
   placeholder?: string;
 }
```

Comment:
`SelectMenuComponentData` should be an union type of every single select manu data

Reply:
I were thinking that I changed it
:/

Reply:
> `SelectMenuComponentData` should be an union type of every single select manu data

isnt it?

Reply:
No
```ts
export interface StringSelectMenuComponentData extends BaseSelectMenuComponentData {
  type: ComponentType.StringSelect;
}

export interface ChannelSelectMenuComponentData extends BaseSelectMenuComponentData {
  type: ComponentType.ChannelSelect;
  channelTypes: ChannelType[];
}

// ...

type SelectMenuComponentData =
  | StringSelectMenuComponentData
  | ChannelSelectMenuComponentData
  | // ...
```

Reply:
ah okey

Reply:
Even SelectMenuComponentData doesnt have to be there